src/IonBinaryReader.ts

```diff
@@ -249,9 +249,14 @@ export class BinaryReader implements Reader {
     private getSymbolString(symbolId: number): string {
         let s: string = null;
         if (symbolId > 0) {
+            if(symbolId > this._symtab.maxId) {
+                throw new Error('Symbol $' + symbolId.toString() + ' greater than maxID.');
+            }
             s = this._symtab.getSymbolText(symbolId);
-            if (typeof (s) == 'undefined') {
-                s = "$" + symbolId.toString();
+            if (typeof (s) === 'undefined') {
```

Comment:
not sure why we're using `typeof` here;  consider changing to `if (s === undefined) {`

Reply:
me either